plugin/core/views.py

```diff
@@ -791,23 +795,34 @@ def format_completion(
         kind = (kind[0], '⚠', "⚠ {} - Deprecated".format(kind[2]))
 
     lsp_label = item["label"]
+    lsp_label_details = item.get("labelDetails")
     lsp_filter_text = item.get("filterText")
     st_annotation = (item.get("detail") or "").replace('\n', ' ')
 
     st_details = ""
     if can_resolve_completion_items or item.get("documentation"):
         st_details += make_command_link("lsp_resolve_docs", "More", {"index": index, "session_name": session_name})
-
-    if lsp_filter_text and lsp_filter_text != lsp_label:
-        st_trigger = lsp_filter_text
-        if st_details:
-            st_details += " | "
-        st_details += "<p>{}</p>".format(html.escape(lsp_label))
+        st_details += " | "
+    if lsp_label_details:
+        lsp_label_detail = lsp_label_details.get("detail")
+        lsp_label_description = lsp_label_details.get("description")
+        st_details += "<p>"
+        # `label` should be rendered most prominent.
+        st_details += _wrap_in_tags("b", lsp_label)
+        if isinstance(lsp_label_detail, str):
+            # `detail` should be rendered less prominent than `label`.
+            # The string is appended directly after `label`, with no additional white space applied.
+            st_details += html.escape(lsp_label_detail)
+        if isinstance(lsp_label_description, str):
+            # `description` should be rendered less prominent than `detail`.
+            # Additional separation is added.
+            st_details += " - {}".format(_wrap_in_tags("i", lsp_label_description))
+        st_details += "</p>"
     else:
-        st_trigger = lsp_label
+        st_details += _wrap_in_tags("p", lsp_label)
```

Comment:
Always putting the `label` in `st_details`, even if it's no different than `lsp_filter_text` is not that great of an idea. It adds unnecessary duplication:

![Screenshot 2021-08-17 at 21 55 53](https://user-images.githubusercontent.com/153197/129791804-18b63caf-0f18-4ef3-bcb1-2568bbb50c83.png)